core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -427,6 +427,10 @@ private function setTimezone()
      */
     private function triggerInitializeSystemHook()
     {
+        if (!is_dir($this->rootDir.'/system/tmp')) {
+            return;
+        }
+
         if (isset($GLOBALS['TL_HOOKS']['initializeSystem']) && \is_array($GLOBALS['TL_HOOKS']['initializeSystem'])) {
```

Comment:
```suggestion
        if (isset($GLOBALS['TL_HOOKS']['initializeSystem']) && \is_array($GLOBALS['TL_HOOKS']['initializeSystem']) && is_dir($this->rootDir.'/system/tmp')) {
```

Reply:
Actually, I am not sure about this. Right now, we are not including the `initialize.php` script if the temp folder has not yet been installed. If we moved the check inside the first if condition, the `initialize.php` file would be included even if the temp folder does not yet exist, wouldn't it?

Reply:
That's correct. However, I would assume there's most likely no such file in an application that's just being installed. But it's possible/likely that there are extensions that use the hook. Also, removing/renaming/disabling the file is easy, removing the extension to circumvent an issue is tedious.

Reply:
See #1690.